pkg/cache/json.go

```diff
@@ -179,6 +192,30 @@ func (q *JSON) computeDigest(fbcFsys fs.FS) (string, error) {
 }
 
 func (q *JSON) Build(ctx context.Context, fbcFsys fs.FS) error {
+	fbc, err := declcfg.LoadFS(ctx, fbcFsys)
+	if err != nil {
+		return err
+	}
+	fbcModel, err := declcfg.ConvertToModel(*fbc)
+	if err != nil {
+		return err
+	}
+
+	if err := q.buildFromModel(fbcModel); err != nil {
+		return err
+	}
+
+	digest, err := q.computeDigest(fbcFsys)
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(filepath.Join(q.baseDir, jsonDigestFile), []byte(digest), jsonCacheModeFile); err != nil {
+		return err
+	}
```

Comment:
I mean from my perspective I'd call `LoadJSONFromModel` with a temporary directory and throw the directory away when I am done.

Reply:
> Would you ever do a integrity check when you're loaded directly from a model?

I don't think so. The digest is a combination of the filesystem contents of the FBC and the cache, so if either change, then the cache is invalid. In this case, we have no FBC filesystem, so the integrity check doesn't make sense.

> I mean from my perspective I'd call LoadJSONFromModel with a temporary directory and throw the directory away when I am done.

Yes, that would be the expected use of this function. But I'm not happy with the API because it _never_ makes sense to do anything other than that. In which case, why ask the caller to pass a directory in? Ideally, the temporary directory would be an implementation detail not exposed in the API. But then how does the caller clean up when they're done? The `Cache` API design _expects_ the cache to be re-usable and saved for later, so there is no `Close()` or `Cleanup()` method in the cache API or in the JSON cache struct.

Maybe need a higher level abstraction that uses a cache under the hood?

Reply:
Well, the "magic" happens in the `packagesFromModel` function so maybe the easier approach is to expose that publicly and then call the functions on the `packageIndex`? This seems to sidestep the caching mechanism entirely if I am not mistaken.

Reply:
Some of those functions accept a cache though (e.g. https://github.com/operator-framework/operator-registry/blob/6c602841934d6e154e38c0574cc140471dc063e6/pkg/cache/pkgs.go#L62)

So right now, `packageIndex` is very much coupled to the cache.

Reply:
Yeah.. I see that now. My mistake.